File: containsDuplicate.py
class Solution:
    def containsDuplicate(self, nums):
        hashset = set()

        for n in nums:
            if n in hashset:
                return True
            hashset.add(n)
        return False

def main():
    nums_input = input("Enter the array of integers separated by spaces: ")
    nums = list(map(int, nums_input.split()))

    solution = Solution()
    result = solution.containsDuplicate(nums)
    
    if result:
        print("True - There are duplicate elements in the array.")
    else:
        print("False - All elements in the array are distinct.")

if __name__ == "__main__":
    main()

File: containsDuplicate.java
import java.util.HashSet;
import java.util.Scanner;

class containsDuplicate {
    public boolean checkForDuplicates(int[] nums) {
        HashSet<Integer> hashset = new HashSet<>();

        for (int n : nums) {
            if (hashset.contains(n)) {
                return true;
            }
            hashset.add(n);
        }
        return false;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the array of integers separated by spaces: ");
        String[] numsInput = scanner.nextLine().split(" ");

        int[] nums = new int[numsInput.length];
        for (int i = 0; i < numsInput.length; i++) {
            nums[i] = Integer.parseInt(numsInput[i]);
        }

        containsDuplicate solution = new containsDuplicate();
        boolean result = solution.checkForDuplicates(nums);

        if (result) {
            System.out.println("True - There are duplicate elements in the array.");
        } else {
            System.out.println("False - All elements in the array are distinct.");
        }
    }
}